treewide: doc: improve compliance with documentation guidelines

mathieuchopstm opened the pull request:
Fixes the following non-ASCII characters usage across tree:
 - Use ASCII single quote (`'`) in place of U+2018 (`‘`) and U+2019 (`’`)
 - Use ASCII hyphen (`-`) in place of U+2013 (`–`)
 - Use ASCII colon (`:`) in place of U+FF1A (`：`)
 - Use ASCII comma (`,`) in place of U+FF0C (`，`)
 - Use ASCII lowercase X letter (`x`) in place of U+00D7 (`×`)
   - on separate commit as this is a debatable change

The (future) documentation guidelines recommend to avoid non-ASCII characters when they are used purely for aesthetic purposes as is the case in files modified here (although some of these may simply come from careless copy-pasting).

All these changes were done mechanically using a simple search-and-replace, with an additional manual cleanup for comma/colon in a few places.

Also fix several issues in various documentation files uncovered by CI when modifying files for the changes. These fixes can be split to a separate PR.

mathieuchopstm committed:
boards: adi: max32{655,690}fthr: replace copyright symbol in board docs

Replace the copyright symbol next to "Arm" with a guidelines-compliant
token. ® is used instead of © to match every other document found in tree:

  $ grep -ri "Arm®" --include=*.rst . | wc -l
  409
  $ grep -ri "Arm©" --include=*.rst . | wc -l
  0

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
doc: release-notes: 3.0: add missing backticks

Add a few missing backticks that caused issues during documentation build.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

etienne-lms reviewed (commented):
Nice cleanup.

I went through the changes and it looks fine. No mess up (i'm sure you did not expect any).

Maybe worth to mention in the commit message body that `I²C` is replaced by plain `I2C`, not using `|sup2|` token (which is a good choice IMHO).



Sections titles, bodies, illustration and tables titles are all rendered nicely (as expected). The TOC looks fine also.

However references do not show rendered tokens, e.g. `|trade|` in plain words here:
(extracted from [build articfacts: kit_pse84_ai/docs/](https://builds.zephyrproject.io/zephyr/pr/98214/docs/boards/infineon/kit_pse84_ai/doc/index.html#programming-and-debugging))
> **Programming and Debugging**
>
> (snip)
>
> Please refer to the [ModusToolbox |trade| software installation guide](https://builds.zephyrproject.io/zephyr/pr/98214/docs/boards/infineon/kit_pse84_ai/doc/index.html#references) to install the Infineon OpenOCD and Edge Protect Security Suite (edgeprotecttools).
>
> (snip)
>
> **References**
> * [PSOC |trade| Edge Arm |reg| Cortex |reg| Multicore SoC Website](https://www.infineon.com/products/microcontroller/32-bit-psoc-arm-cortex/32-bit-psoc-edge-arm/psoc-edge-e84#Overview)

Ditto for console logs (of course). E.g. here:
(extracted from [build articfacts: niosv_g/docs/](https://builds.zephyrproject.io/zephyr/pr/98214/docs/boards/intel/niosv_g/doc/index.html#overview))
> **Overview**
>
> niosv_g board is based on Intel FPGA Design Store Nios® V/g Hello World Example Design system and this complete system is consisted of following IP blocks:
> ```
> Nios |reg| V/g Processor Intel |reg| FPGA IP
> JTAG UART Intel |reg| FPGA IP
> On-Chip Memory Intel |reg| FPGA IP
> ```

Last, *boards/intel/niosv_g/doc/index.rst* and *boards/intel/niosv_m/doc/index.rst* have some other non-rendered |trade| tokens but that's because `Note:` section is mistakenly formated as a `.. code-block:: console` section. E.g. [here](https://builds.zephyrproject.io/zephyr/pr/98214/docs/boards/intel/niosv_g/doc/index.html#create-nios-reg-v-g-processor-example-design-system-in-fpga).

etienne-lms committed:
boards: infineon: kit_pse84_eval: fix QSPI spelling in doc

Fix typo in kit_pse84_eval board documentation replacing QSI| with
QSPI.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented:
Hi @mathieuchopstm, I've appended with other related fixes to your series.
Feel free to do whatever you want with. If you prefer I move them to a specific P-R, please tell me.

mathieuchopstm commented:
> > Summoning the documentation guru @kartben for help about how those should be handled (keep the non-ASCII characters? some Sphinx magic to have the replacement happen forcefully?)
> 
> Well yeah, given it's 2025, I really think we should embrace non-ASCII characters (or at least define an allowlist of accepted common ones like ™,  © , etc.), at least for documentation files.

In this case, should the current replacements (`|copy|`, `|trade|`, etc) be "upgraded" to Unicode characters? I can open a concurrent PR doing treewide cleanup in this direction instead (+ guidelines update) if so.

kartben commented:
Sorry that I never replied to the hanging question @mathieuchopstm. I think I lean towards "unconditionally use non-ASCII characters instead of the substitutions?" basically for the reason mentioned in my previous comment. I would basically turn our existing set of substitutions into the "allowlist" of non-ASCII characters that are just fine in .rst documentation

FWIW the Linux community seems to have no issue with non-ASCII characters, from what I can see, or at least they don't have tooling to actively prevent them from making it into the codebase.

mathieuchopstm commented:
Based on ArchWG feedback, I will repurpose this PR to perform only the "part 1" (replacing cosmetic Unicode characters with their ASCII counterparts), and drop the "part 2" (replace TM/(R)/etc with reST `|substitutions|`)

This would match the new guidelines proposed in #102996:
> Avoid adding non-ASCII characters purely for aesthetic purposes.

mathieuchopstm committed:
doc: treewide: replace usage of several cosmectic non-ASCII characters

Fixes the following non-ASCII characters usage across tree:
 - Use ASCII single quote (') in place of U+2018 (‘) and U+2019 (’)
 - Use ASCII hyphen (-) in place of U+2013 (–)
 - Use ASCII colon (:) in place of U+FF1A (：)
 - Use ASCII comma (,) in place of U+FF0C (，)

The documentation guidelines recommend to avoid non-ASCII characters when
they are used purely for aesthetic purposes as is the case here (although
some of these may simply come from careless copy-pasting).

All these changes were done mecanically using a simple search-and-replace,
with an additional manual cleanup for comma/colon in a few places.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

mathieuchopstm committed:
doc: treewide: replace U+00D7 with ASCII lowercase X

Use ASCII lowercase X letter (x) in place of U+00D7 (×) as that character
is used purely for aesthetic reasons, which goes against the documentation
guidelines.

Changes done mecanically using a simple search-and-replace.

Signed-off-by: Mathieu Choplain <mathieu.choplain-ext@st.com>

etienne-lms reviewed (commented):
I think most of the UTF characters introduced in the boards documentation come from copy/paste of some external document content. I guess new occurrences will be added in the future. Should the CI emit some warning (non-blocking) messages on occurrences for very common characters that have ASCII equivalents? (I guess there are several code ranges to consider). Or maybe having an script in Zephyr **scripts/** to help one to perform such replacement.

Otherwise LGTM but regarding the release note files.

kartben commented on boards/adi/eval_adin2111d1z/doc/index.rst:
```diff
@@ -59,7 +59,7 @@ Hardware
 | 6 pin header                          | SPI pins for ADIN2111                        | Only for testing                                                                                                                                                                 |
 +---------------------------------------+----------------------------------------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
-| BLE antenna (Kyocera M310220)         | 2.4 GHz on-board antenna                     | Freq. range 2400 –2485 MHz; G(PEAK) 1.7 dBi; linearly polarized.                                                                                                                 |
+| BLE antenna (Kyocera M310220)         | 2.4 GHz on-board antenna                     | Freq. range 2400 -2485 MHz; G(PEAK) 1.7 dBi; linearly polarized.                                                                                                                 |
```
```suggestion
| BLE antenna (Kyocera M310220)         | 2.4 GHz on-board antenna                     | Freq. range 2400–2485 MHz; G(PEAK) 1.7 dBi; linearly polarized.                                                                                                                 |
```

kartben commented on boards/adi/max32662evkit/doc/index.rst:
```diff
@@ -28,7 +28,7 @@ Hardware
     - Internal Low-Power 7.3728MHz
     - Ultra-Low-Power 80kHz
-    - 16MHz–32MHz (External Crystal Required)
+    - 16MHz-32MHz (External Crystal Required)
```
> Use ASCII hyphen (-) in place of U+2013 (–)

en dash and em dash are two examples of symbols that I would really hope we can keep for "conventional typography", as per the guidelines.

Here, an en dash should be fine, and even encouraged IMO. https://www.chicagomanualofstyle.org/qanda/data/faq/topics/HyphensEnDashesEmDashes/faq0002.html#:~:text=And%20in%20fact%20en%20dashes%20specify%20any%20kind%20of%20range

Bonus points if you have a change to come up with a magical search and replace to not repeat units (but like someone pointed in the review, these will keep coming back if they're copy-pasted from external docs...

```suggestion
    - 16–32MHz (External Crystal Required)
```

kartben reviewed (commented):
can you please try to keep th PR description up-to-date as the actual list of changes evolves? Thanks for this! This is a partial review only commenting on dashes which I would hope we can keep using (wherever they're used correctly, that is)

jadonk reviewed (commented):
I cannot provide overall feedback, but for the 2 fixes this did on Beagle docs, it looks good, replacing ’ (UTF-8 E2 80 99) with ' (UTF-8 27).
